src/systemcmds/serial_test/serial_test.c

```diff
@@ -598,7 +598,7 @@ static void process_write_data(struct g_mod_t *g_mod, struct cli_args_t *g_cl)
 		count += c;
 
 		if (c <= g_mod->_write_size) {
-			g_mod->_write_count_value = g_mod->_write_data[c];
+			g_mod->_write_count_value = next_count_value(g_mod->_write_data[c - 1], g_cl->_ascii_range);
```

Comment:
`c` could be 0.

Reply:
Amazing analytics skills. Thanks for pointing to this.